sensors/nxt_i2c_sensor_core.c

```diff
@@ -377,6 +399,18 @@ static int nxt_i2c_sensor_detect(struct i2c_client *client,
 #endif
 
 	/*
+	 * NXT Temperature sensor detection is handled here.
+	 */
+	if (client->addr == 0x4c) {
+		ret = nxt_i2c_sensor_detect_lego_temp(client);
+		if (ret)
```

Comment:
I don't see how the second one works since by the time we have entered `nxt_i2c_sensor_detect_lego_temp` we are committed to the belief that the device is the temperature sensor. The only return values we can get is a failure from the write, `-ENODEV`or 0. A failure from the write seems like something that would indicate more than just "this is not the temp sensor, but it might be something else" since it's at address 0x1, which presumably the vast majority of devices would map.